### PR TITLE
Prompt user when closing tab or when navigating away.

### LINKED ISSUE #588: Prompt user from leaving team assignment with unsaved changes
When a user has made changes in the `Assign Practitioners` view, they should be prompted with an alert to save or discard their changes.

- [x] Add an alert with message: `Unsaved Changes: please Save or Discard changes made [team name]
- [x] Add a `Discard Changes` button which resets the practitioner list and dropdown and does not navigate away from the page

### PR DESCRIPTION
**Pull request by p-netm**
Adds prompts for :
- There is unsaved changes on the practitioner assign page:
   - when the user tries navigating to another page
   - when the user tries closing the browser window/tab

closes #588

### CODE REVIEW

**Comment by p-netm** [comment]
Created #607 to track one cleanup issue for this: use of any and tests for the hook, I feel like testing this falls out of the usual jest wrapper approach.

File: src/constants.tsx [hunk 1]
```diff
@@ -381,3 +381,8 @@ export const PRACTITIONER_CODE = {
   text: 'Community Health Worker',
 };
 export type PRACTITIONER_CODE = typeof PRACTITIONER_CODE;
+
+// prompt message
+export const ON_REROUTE_WITH_UNSAVED_CHANGES =
+  'Rerouting will discard any unsaved changes, confirm';
```
Comment (cKellyDesign) [inline src/constants.tsx]: `..., please confirm`

**Comment by p-netm** [comment]
@cKellyDesign Should we consider adding a reset button to the assign practitioner page.
also i am lost of ideas on how to test this,

**Comment by cKellyDesign** [comment]
@p-netm no worries, let's not let that block getting this out, I've added a ticket for adding this test https://github.com/onaio/reveal-frontend/issues/609

**Pull request merged**